share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -33,7 +33,7 @@ default:
   artifacts:
     paths:
       - "${CI_PROJECT_DIR}/jobs_scratch_dir"
-  tags: ["spack", "public", "medium", "x86_64"]
+  tags: ["spack", "public", "large", "x86_64"]
```

Comment:
The locking issue, which I'll try to investigate next, happened consistently in CI when I scaled to the "large" instances - so I have no numbers on those. The numbers above refer to my dev laptop. I'll try to check carefully what might be causing the issue, since having spurious deadlocks is not funny at all.